### PR TITLE
Implement method to retrieve battery value

### LINKED ISSUE #14: As a user, I want to see battery % (approx)
- [ ] Add ui to display bettery %
- [ ] Apply typical Lipo discharge profile LUT to translate Vto%

### PR DESCRIPTION
**Pull request by fustilio**
Returned value is an integer between from 0-100 corresponding to 3100mV to 4000mV.

Usage:
`TS_HAL.lcd_printf("Battery: %d%%  ", TS_HAL.power_get_batt_level(), NULL, NULL);`

### CODE REVIEW

File: alpha/hal.cpp [hunk 1]
```diff
@@ -304,6 +307,16 @@ void _TS_HAL::power_set_mic(bool enabled)
   EXIT_CRITICAL;
 }
 
+uint8_t _TS_HAL::power_get_batt_level(){
+  long level;
+  ENTER_CRITICAL;
+#ifdef HAL_M5STICK_C
+  level = constrain(map(M5.Axp.GetVbatData() * 1.1, 3100, 4000, 0, 100), 0, 100);
```
Comment (jamesi8086) [inline alpha/hal.cpp]: put the constrain call outside the critical section

Comment (fustilio) [reply alpha/hal.cpp]: Fixed!

File: alpha/hal.cpp [hunk 2]
```diff
@@ -36,6 +36,9 @@ void _TS_HAL::begin()
   M5.Lcd.fillScreen(BLACK);
   M5.Lcd.setTextSize(1);
 
+  // enable to measure battery level
+  M5.Axp.EnableCoulombcounter();
```
Comment (jamesi8086) [inline alpha/hal.cpp]: if this is required to measure battery level , maybe we just need to enable it right before we measure, and disable it after?
But battery voltage measurement isn't really the job of a coulomb counter though. Its to measure charges sent/recv from batt.

**Review by jamesi8086 — APPROVED** [review]

File: alpha/hal.cpp [hunk 3]
```diff
@@ -304,6 +307,16 @@ void _TS_HAL::power_set_mic(bool enabled)
   EXIT_CRITICAL;
 }
 
+uint8_t _TS_HAL::power_get_batt_level(){
+  long level;
+  ENTER_CRITICAL;
+#ifdef HAL_M5STICK_C
+  level = map(M5.Axp.GetVbatData() * 1.1, 3100, 4000, 0, 100);
```
Comment (jamesi8086) [inline alpha/hal.cpp]: only M5.Axp.GetVbatData() should be within CS, map should be outside.

**Review by jamesi8086 — APPROVED** [review]
fix minor issues before merge

**Pull request merged**